pkg/controller/core/workload_controller.go

```diff
@@ -152,10 +168,35 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 	log.V(2).Info("Reconcile Workload")
 
 	if len(wl.OwnerReferences) == 0 && !wl.DeletionTimestamp.IsZero() {
+		// manual deletion triggered by the user
 		return ctrl.Result{}, workload.RemoveFinalizer(ctx, r.client, &wl)
 	}
 
-	if apimeta.IsStatusConditionTrue(wl.Status.Conditions, kueue.WorkloadFinished) {
+	finishedCond := apimeta.FindStatusCondition(wl.Status.Conditions, kueue.WorkloadFinished)
+	if finishedCond != nil && finishedCond.Status == metav1.ConditionTrue {
+		if !features.Enabled(features.ObjectRetentionPolicies) || r.workloadRetention.afterFinished == nil {
+			return ctrl.Result{}, nil
+		}
+
```

Comment:
nit, I would suggest to drop the empty lines in this block. I don't think they really improve readability, and it is not so consistent with the style in the codebase. Yes, sometimes we use empty lines to emphasize different stages, but in this case there is an empty line almost after every line of code. This is too much.

Reply:
Surely, have left only divisions between three main blocks here

Reply:
Cool, drop also the line just before the last `return ctrl.Result{}, nil`

Reply:
Sure, done